src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -134,12 +142,14 @@ public void setPeriod(String period) {
     public void visitJavadocToken(DetailNode ast) {
```

Comment:
logic is still over complicated.

you need to have visitJavadocToken method to be like:
```
String firstSentence = getFirstSentence(ast, period);
if (firstSentence.empty) {
   log(MSG_SUMMARY_JAVADOC_MISSING)
} else if (! isValid(firstSentence)) {
   log(MSG_SUMMARY_FIRST_SENTENCE)
}
else {
   if (containsForbiddenFragment(firstSentence)) { .....}
}
```